File: net472/webapi/BuildableWebApi/BuildableWebApi/Controllers/HomeController.cs
using System.Configuration;
using System.Runtime.Caching;
using System.Web.Configuration;
using System.Web.Mvc;

namespace BuildableWebApi.Controllers
{
    public class HomeController : Controller
    {
        public ActionResult Index()
        {
            ViewBag.Title = "Home Page";
            var cache = new MemoryCache("MemoryCache");
            GetConfigurationValues();
            return View();
        }

        private void GetConfigurationValues()
        {
            string constConnectionString = "FirstConnectionString";
            string constAppSetting = "ClientValidationEnabled";

            var conn = ConfigurationManager.ConnectionStrings["FirstConnectionString"];
            var conn2 = ConfigurationManager.ConnectionStrings["SecondConnectionString"].ConnectionString;
            var conn3 = ConfigurationManager.ConnectionStrings[constConnectionString].ConnectionString;

            var webConn1 = WebConfigurationManager.ConnectionStrings["FirstConnectionString"];
            var webConn2 = WebConfigurationManager.ConnectionStrings["SecondConnectionString"].ConnectionString;
            var webConn3 = WebConfigurationManager.ConnectionStrings[constConnectionString].ConnectionString;

            var appSetting = ConfigurationManager.AppSettings["ClientValidationEnabled"];
            var appSetting3 = WebConfigurationManager.AppSettings[constAppSetting];

        }
    }
}

File: net472/VBBuildableWebApi/VBWebApi/Controllers/HomeController.vb
Imports System.Configuration
Imports System.Runtime.Caching
Imports System.Web.Configuration 
Imports System.Web.Mvc

Public Class HomeController
    Inherits System.Web.Mvc.Controller

    Function Index() As ActionResult
        ViewData("Title") = "Home Page"
        dim cache as MemoryCache = new MemoryCache("MemoryCache")

        GetConfigurationValues()

        Return View()
    End Function

    Private Sub GetConfigurationValues() 
        dim constConnectionString As String = "FirstConnectionString"
        Dim constAppSetting as String = "ClientValidationEnabled"

        dim conn = ConfigurationManager.ConnectionStrings("FirstConnectionString")
        dim conn2 = ConfigurationManager.ConnectionStrings("SecondConnectionString").ConnectionString
        dim conn3 = ConfigurationManager.ConnectionStrings(constConnectionString).ConnectionString

        dim webConn1 = WebConfigurationManager.ConnectionStrings("FirstConnectionString")
        dim webConn2 = WebConfigurationManager.ConnectionStrings("SecondConnectionString").ConnectionString
        dim webConn3 = WebConfigurationManager.ConnectionStrings(constConnectionString).ConnectionString

        dim appSetting = ConfigurationManager.AppSettings("ClientValidationEnabled")
        dim appSetting3 = WebConfigurationManager.AppSettings(constAppSetting)

    End Sub

End Class
